### ISSUE TITLE
PHP 7 compatibility

### ISSUE THREAD

**Issue by gplanchat** [akeneo/pim-community-dev#3343]
There are some classes using new reserved names in the PIM source code which will not be compatible with PHP 7.0 :

```Fatal error: Cannot use Pim\Bundle\CatalogBundle\Validator\Constraints\String as String because 'String' is a special class name in /Users/gplanchat/Luni/projects/pim-akeneo/vendor/akeneo/pim-community-dev/src/Pim/Bundle/CatalogBundle/Validator/ConstraintGuesser/StringGuesser.php on line 8```

I only found class `Symfony\Component\Validator\Constraint\String` for now, maybe there are some others

**Comment by jjanvier** [comment]
Hello Gregory @gplanchat, 

Thanks for raising the issue ;)

We are not yet compatible with PHP7 but we are not so far actually. Maybe you tried the same tool that us to detect those incompatibilities https://github.com/sstalle/php7cc ?

```
jjanvier:php7cc$ php bin/php7cc.php ~/akeneo/pim_master/pcd/src                                                                             

File: /home/jjanvier/akeneo/pim_master/pcd/src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/CategoryRepository.php
Line 309. Possible array element creation during by-reference assignment: $tree[$node->getId()] =& $vectorMap[$node->getId()];
Line 338. Possible array element creation during by-reference assignment: $tree[$nodeId] =& $vectorMap[$nodeId];


File: /home/jjanvier/akeneo/pim_master/pcd/src/Pim/Bundle/CatalogBundle/Validator/Constraints/Numeric.php
Line 14. Reserved word "numeric" used as a class, interface or trait name: /**
 * Constraint
 *
 * @author    Antoine Guigan <antoine@akeneo.com>
 * @copyright 2013 Akeneo SAS (http://www.akeneo.com)
 * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
 */
class Numeric extends Constraint
{
}


File: /home/jjanvier/akeneo/pim_master/pcd/src/Pim/Bundle/CatalogBundle/Validator/Constraints/String.php
Line 14. Reserved word "string" used as a class, interface or trait name: /**
 * Constraint
 *
 * @author    JM Leroux <jean-marie.leroux@akeneo.com>
 * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
 * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
 */
class String extends Constraint
{
}


File: /home/jjanvier/akeneo/pim_master/pcd/src/Pim/Bundle/TransformBundle/Tests/Unit/Transformer/Guesser/ChainedGuesserTest.php
Line 50. Function argument(s) returned by "func_get_args" might have been modified: func_get_args();

Checked 1473 file(s) in 21.653302 second(s)
```

We were planning to fix those issues in the coming 1.5 version. Feel free to open a PR to rename the constratints and the validators to `IsString` and `IsNumeric`. We'll do it in the coming weeks otherwise.


Thanks ;)

**Comment by nidup** [comment]
@gplanchat thanks again for raising the issue!

More info here https://medium.com/akeneo-labs/akeneo-pim-php7-and-hhvm-compatibility-98e7fe0ce582
